### PR TITLE
New attendance forms

### PR DESCRIPTION
**Pull request by zshipchandler**
added new backend endpoints for checkin forms, excused absence forms, and took out hardcoded last meeting variables needed for the donut and horizontal bar graph on dashboard.

also reorganized files & debugged other endpoints

### CODE REVIEW

File: backend/services/meetingHistory.js [hunk 1]
```diff
@@ -23,8 +23,9 @@ module.exports = async function getMeetingHistory() {
             meetingType: page.properties.MeetingType.title[0].text.content,
             meetingNumber: page.properties['Meeting #'].number,
             numAttended: page.properties['Attended'].number,
-            numAttended: page.properties['# Unexcused Absences'].number,
-            attendancePercent: page.properties['%'].formula.number
+            numAbsent: page.properties['# Unexcused Absences'].number,
```
Comment (Causality-C) [inline backend/services/meetingHistory.js:26]: Nit: not sure if using # or % would be a good idea here from a style perspective, we generally use those symbols for other things (ie comments or string formatting)

File: formPage/my-app/src/CheckinForm.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+async function getCheckin(state) {
+    const request = await fetch("http://localhost:5000/updateCheckin", {
```
Comment (Causality-C) [inline formPage/my-app/src/CheckinForm.js:4]: You can make API calls (POST, GET, PUT, etc) functions that take in an URL and body so you can avoid re-writing this boiler plate code (ie. https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)

Also, does the API work here if you don't supply cors?

File: formPage/my-app/src/ExcusedAbsenceForm.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      });
+      // Get Status and Response
+      const resp = await request.json();
+      const msg = await resp.msg;
+      return msg;
+        
+}
+
+
+export default class ExcusedAbsenceForm extends React.Component {
```
Comment (Causality-C) [inline formPage/my-app/src/ExcusedAbsenceForm.js:23]: Let's make a decision to use class-based or function-based components throughout the entire project. Is there a reason why we use class based here?

File: frontend/my-app/src/components/BarChartHorizontalDashboard.js [hunk 4]
```diff
@@ -10,15 +10,10 @@ ChartJS.register(
...
-    //hardcoded for now
-    var lastGen = 2;
-    var lastEng = 2;
-    var lastDes = 2;
-    var lastProd = 2;
 
     useEffect(() => {
      const fetchData = async () => {
-        await fetch("http://localhost:5000/BarChartHorizData?lastGen="+lastGen+"&lastEng="+lastEng+"&lastDes="+lastDes+"&lastProd="+lastProd, {
+        await fetch("http://localhost:5000/BarChartHorizData", {
```
Comment (Causality-C) [inline frontend/my-app/src/components/BarChartHorizontalDashboard.js:16]: Great change!

File: frontend/my-app/src/components/DonutChart.js [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
-
-    //should get this from automated meeting number, hardcoded for now
-    const lastGenMeetingNum = 2;
-
-
     const [chart, setChart] = useState();
 
     const fetchData = async () => {
-        await fetch("http://localhost:5000/DonutData?type=" + props.type + "&last=" + lastGenMeetingNum, {
+        await fetch("http://localhost:5000/DonutData?type=" + props.type, {
```
Comment (Causality-C) [inline frontend/my-app/src/components/DonutChart.js:11]: You can do string interpolation here like

\`http://localhost:5000/DonutData?type=${props.type}`

File: frontend/my-app/src/pages/teamPage.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import {Card} from 'react-bootstrap'
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+
+import './teamPage.css';
+
+const TeamPage = (props) => {
+  return (
+    <div>
+
```
Comment (Causality-C) [inline frontend/my-app/src/pages/teamPage.js]: Any reason for these random three spaces?

Also, may be worth using a linter (prettier) in the future to standardize code styling

**Comment by Causality-C** [comment]
Besides the comments I mentioned above, LGTM, is there anything you want me to look at in particular? Is everything working correctly?

File: backend/z_index.js [hunk 7]
```diff
@@ -273,6 +276,68 @@ app.post('/meeting/:meetingType', async (req, res) => {
...
+  else if (meetingType == "Product") {
+    pageId = process.env.NOTION_PRODUCT_MEETING_INFO;
+  }
+  else if (meetingType == "Design") {
+    pageId = process.env.NOTION_DESIGN_MEETING_INFO;
+  }
+  const response = await notion.pages.retrieve({ page_id: pageId });
+  var activeMeeting = false;
+  // is an empty array if manually deleted otherwise an empty string
+  if (response.properties['Meeting Code'].title.length && response.properties['Meeting Code'].title[0].plain_text !== '') {
```
Comment (evanzlock) [inline backend/z_index.js]: I think we should check if Date.now() < response.properties['End Time'].number instead to set the active meeting setting.  My bad on this cause I had this logic before in my code.

**Comment by evanzlock** [comment]
Wow great job this week on making so much progress :) Just left one comment but everything else looks super good! Thanks for all the detailed comments too!

**Pull request merged**